compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -609,7 +612,12 @@ impl<'context> Elaborator<'context> {
                 has_inline_attribute,
                 type_span,
             );
-            let pattern = self.elaborate_pattern(pattern, typ.clone(), DefinitionKind::Local(None));
+            let pattern = self.elaborate_pattern_and_store_ids(
+                pattern,
+                typ.clone(),
+                DefinitionKind::Local(None),
+                &mut parameter_idents,
```

Comment:
It feels like it may be cleaner to just save the hir idents on `self` then it can cleared after we are done elaborating a function meta. We can then remove the ` #[allow(clippy::too_many_arguments)]` and the threading of the mutable reference.

Reply:
I want to avoid cluttering `self` with too many fields that are only used temporarily in a couple functions. Otherwise the object grows even larger, and it's already quite large since it has all of the name resolver's and type checker's fields.